tests/optimize/test_objective.py

```diff
@@ -74,5 +78,152 @@ def tearDown(self):
+        r6_inv = np.power(0.9*1./rs, 6)
+        gs = np.exp(-(1/1.5)*4.*1.0*(r6_inv**2 - r6_inv))
+        sim_rdf = relentless._math.Interpolator(rs,gs)
+
+        rs = np.arange(0.05,5.0,0.1)
+        r6_inv = np.power(0.9/rs, 6)
+        gs = np.exp(-4.*1.0*(r6_inv**2 - r6_inv))
+        tgt_rdf = relentless._math.Interpolator(rs,gs)
+
+        rs = np.arange(0,3.6,3.6/1000)[1:]
+        r6_inv = np.power(0.9*1./rs, 6)
+        if var is self.epsilon:
+            dus = 4*(r6_inv**2 - r6_inv)
+        elif var is self.sigma:
+            dus = (48.*1.0/0.9)*(r6_inv**2 - 0.5*r6_inv)
+        dudvar = relentless._math.Interpolator(rs,dus)
+
+        if ext:
+            norm_factor = 1.
+        else:
+            norm_factor = 1000.
+        sim_factor = 50**2*(1/1.5)/(1000*norm_factor)
+        tgt_factor = 50**2*(1/1.5)/(1000*norm_factor)
+
+        r = np.arange(0.05, 3.55, 0.00355)
+        y = 2*np.pi*r**2*(sim_factor*sim_rdf(r)-tgt_factor*tgt_rdf(r))*dudvar(r)
+        return scipy.integrate.trapz(x=r, y=y)
+
+    def test_init(self):
+        relent = relentless.optimize.RelativeEntropy(self.target,
+                                                     self.simulation,
+                                                     self.potentials,
+                                                     self.thermo)
+        self.assertEqual(relent.target, self.target)
+        self.assertEqual(relent.simulation, self.simulation)
+        self.assertEqual(relent.potentials, self.potentials)
+        self.assertEqual(relent.thermo, self.thermo)
+        self.assertIsNone(relent.communicator)
+
+        #test communicator argument
+        comm = relentless.mpi.world
+        relent.communicator = comm
+        self.assertEqual(relent.target, self.target)
+        self.assertEqual(relent.simulation, self.simulation)
+        self.assertEqual(relent.potentials, self.potentials)
+        self.assertEqual(relent.thermo, self.thermo)
+        self.assertEqual(relent.communicator, comm)
+
+        #test invalid target ensemble
+        with self.assertRaises(ValueError):
+            relent.target = relentless.ensemble.Ensemble(T=1.5, P=1, N={'1':50})
+
+    def test_compute(self):
+        """Test compute method"""
+        relent = relentless.optimize.RelativeEntropy(self.target,
+                                                     self.simulation,
+                                                     self.potentials,
+                                                     self.thermo)
+
+        res = relent.compute()
+        self.assertIsNone(res.value)
+        grad_eps = self.relent_grad(self.epsilon)
+        grad_sig = self.relent_grad(self.sigma)
+        err_grad_eps = 1.-np.abs(grad_eps/res.gradient[self.epsilon])
+        err_grad_sig = 1.-np.abs(grad_sig/res.gradient[self.sigma])
+        self.assertLess(err_grad_eps, 0.01) #<1% error in floating point rounding
+        self.assertLess(err_grad_sig, 0.01)
```

Comment:
If you want to get agreement to a certain number of decimal places, use the `places` option. For example, if the numbers are order 1, 1% would be 2 places:
```suggestion
        self.assertAlmostEqual(res.gradient[self.epsilon], grad_eps, places=2)
        self.assertAlmostEqual(res.gradient[self.sigma], grad_sig, places=2)
```
You might want to make places smaller depending on what the values of these is, though.

There is also numpy.testing.assert_allclose, which will let you set an atol and rtol for closeness, if you prefer that route.